src/test/scala/chiselTests/DataEqualitySpec.scala

```diff
@@ -22,7 +22,9 @@ class EqualityTester(lhsGen: => Data, rhsGen: => Data) extends BasicTester {
 
   assert(equalityModule.out)
 
-  stop()
+  when(RegNext(next = true.B, init = false.B)) {
+    stop()
+  }
```

Comment:
This is necessary because this test relies on the ordering of asserts and stops. With this PR, the assert and stop will be in separate always blocks which have no defined execution order. I expect that because `stop` goes into the `Verification` layer and `assert` goes into the `Verification.Assert` layer and that these are then bound in with `Verification` first and then `Verification.Assert` that a simulator will likely evaluate the `stop` before the `assert`. (There is obviously no guarantee of this, though.)

Adding an explicit delay to the `stop` of one cycle avoids this problem.

(This test was bad originally at it relied on assumptions about ordering of side effects which moving things into layers exposed.)